### ISSUE TITLE
First placement of rule of Jupyter+Github

### ISSUE THREAD

**Issue by jpotvin** [Xalgorithms/xadf-active-repositories#2]
We have two kinds of things to maintain: IoR rule notebooks, and reference data tables.  One could say they're the same thing.

There's an operational question to step Consider a reference data table, for example. The "Distance Register" for the gas stations.

It's not part of "the rule", except that the rule needs to draw upon it.

After some thought I concluded this table (and others like it) deserves its own Jupyter/Zeppelin notebook. This way the maintainer can explain its nuances (e.g. how "distance" should be determined). But this table and its particular documentation also need to be versioned, so I'd want the author to keep it on Github, Bitbucket or equivalent.

Here are the two key "how to" texts for keeping Jupyter notebooks on Github:

            https://help.github.com/articles/working-with-jupyter-notebook-files-on-github/
            https://gist.github.com/pbugnion/ea2797393033b54674af
http://pascalbugnion.net/blog/ipython-notebooks-and-git.html

At this point I THINK I should drop each single table file in csv format into https://gist.github.com  ... do you agree?
        See: https://help.github.com/articles/about-gists/
And then I'd pull that versioned table into the Jupyter notebook... er, am I overcomplicating things??

I can call one notebook from another, this way:
https://nbviewer.jupyter.org/github/ipython/ipython/blob/master/examples/IPython%20Kernel/Importing%20Notebooks.ipynb

See also https://forums.databricks.com/questions/154/can-i-run-one-notebook-from-another-notebook.html

But I'm not sure if my rule notebook should call reference data tables from other notebooks, or from Github.

Meanwhile, when maintaining a Jupyter notebook in Github/Gist, Github won't render the executable parts, but to have the executable results viewable from the Github-stored files, there is https://nbviewer.jupyter.org/

In order to proceed, I need some feedback from Don and/or Hayk.

**Comment by karfai** [comment]
As we've been doing, organize them in https://github.com/Xalgorithms/lichen-examples for now. Once there's a little bit of data in there, we'll get an idea how to manage. If you can, you can drop notebooks in there too.

**Comment by jpotvin** [comment]
Will do -- sorry had to attend to other stuff most of today. (One of the
reasons I start very early.)


Joseph Potvin
jpotvin@opman.ca
@jrpotvin
joseph.potvin@letsgomoose.com
jpotvin@xalgorithms.org
Mobile: 819-593-5983

**Comment by mcr** [comment]
I have read https://nbviewer.jupyter.org/ and
https://jupyter.readthedocs.io/en/latest/index.html
and I have no idea what a jupyter notebook.

Why did they explain what IPython REPL is to me?

--
]               Never tell me the odds!                 | ipv6 mesh networks [
]   Michael Richardson, Sandelman Software Works        | network architect  [
]     mcr@sandelman.ca  http://www.sandelman.ca/        |   ruby on rails    [

**Comment by karfai** [comment]
tldr => Jupyter (ipython) is a markdown page with "live" python code running in it. Used often in the "data analyst" space.
